include/aie/Dialect/AIE/IR/AIEOps.td

```diff
@@ -741,44 +741,66 @@ def AIE_DMABDPACKETOp: AIE_Op<"dma_bd_packet", []> {
   }];
 }
 
-def AIE_DMABDOp: AIE_Op<"dma_bd", []> {
-  let summary = "Declare a dma block descriptor op";
+def AIE_DMABDOp: AIE_Op<"dma_bd", [
+    ParentOneOf<["MemOp", "MemTileDMAOp", "ShimDMAOp", "DMAOp"]>,
+  ]> {
+  let summary = "Declare a dma buffer descriptor op";
   let description = [{
-    This operation describes a block descriptor for DMA operations. In particular, it specifies
-    what buffer addresss to use, the transfer length, and the buffer type (A or B).
+    This operation describes a buffer descriptor for DMA operations. In particular, it specifies
+    what buffer to use, and optionally:
 
-    This operation must be used in an MLIR block that lives inside a MemOp's region.
-    The block descriptor specifies what lock to use and the buffer configuration.
+        1. the offset into the buffer;
+        2. the transfer length;
+        3. the sizes and strides for n-d tensor addressing (described below);
+        4. the "bd_id" with which to associate the buffer descriptor (most often left empty).
+
+    `offset`, `len`, `size`s and `stride`s are all denominated in element width; e.g., transferring the whole of
```

Comment:
is it in 32b multiples or is it in element with multiples (where element width changes depending on the type)?

Reply:
or on line 758, does len need to be equal to 512 in both cases?

Reply:
it's "element width multiples" - line 758 is correct. this PR allows/enables the user to (almost) only think in terms of element width, forgetting 32b alignment except in one place - the final stride dim, which does have to be a multiple of 32b. I will update this doc string to say mention that (the last stride dim).

Reply:
still waking up...